src/v1/combo-chart/combo-chart.ts

```diff
@@ -0,0 +1,669 @@
+  "#228B22",
+  "#008080",
+  "#FF6347",
+  "#40E0D0",
+  "#800080",
+  "#2E8B57",
+  "#FF8C00",
+  "#8B4513",
+  "#008B8B",
+  "#FF69B4",
+  "#7CFC00",
+  "#4169E1",
+  "#FF1493",
+  "#ADFF2F",
+  "#800000",
+  "#20B2AA",
+  "#F08080",
+];
+
+const COMBO_CHART_TYPE = {
+  bar: "bar",
+  line: "line",
+  stack: "bar-stack",
+  scatter: "scatter",
+};
+
+function getDataForColumn(column: ChartColumn, dataArr: DataPointsArray) {
+  const idx = _.findIndex(dataArr.columns, (colId) => column.id === colId);
+  return _.uniq(_.map(dataArr.dataValue, (row) => row[idx]));
+}
+
+function getSeriesDataForColumn(column: ChartColumn, dataArr: DataPointsArray) {
+  const idx = _.findIndex(dataArr.columns, (colId) => column.id === colId);
+  return _.uniq(_.map(dataArr.dataValue, (row) => row[idx]));
+}
+
+function getDataForColumnForSeries(
+  column: ChartColumn,
+  dataArr: DataPointsArray,
+  col: any
+) {
+  const idx = _.findIndex(dataArr.columns, (colId) => column.id === colId);
+  return dataArr.dataValue
+    .filter((item) => item[2] === col)
+    .map((row) => row[idx]);
+}
+
+function getChartDataModel(
+  configDimensions,
+  dataArr: DataPointsArray,
+  inputType
+) {
+  const xAxisColumns = configDimensions?.[0].columns ?? [];
+  const yAxisColumns = configDimensions?.[1].columns ?? [];
+  const legend = configDimensions?.[2]?.columns[0] ?? [];
+  const type = inputType.split("-")[0];
+  const isStacked = inputType.split("-").length > 1;
+
+  return {
+    getLabels: () => getDataForColumn(xAxisColumns[0], dataArr),
+    getDatasets: () => {
+      if (!_.isEmpty(legend)) {
+        return _.map(getSeriesDataForColumn(legend, dataArr), (col, idx) => {
+          return {
+            label: `${col}- ${inputType}`,
+            data: getDataForColumnForSeries(yAxisColumns[0], dataArr, col),
+            type: `${type}`,
+            // yAxisID: `${type}-y${idx.toString()}`,
+            stack: `${type}-x0${isStacked ? "-stacked" : "y" + idx.toString()}`,
+            yPos: idx,
+            backgroundColor: '',
+            borderColor: '',
+            datalabels: {
+              anchor: "end",
+            },
+          };
+        });
+      } else {
+        return _.map(yAxisColumns, (col, idx) => {
+          return {
+            label: `${col.name}- ${inputType}`,
+            data: getDataForColumn(col, dataArr),
+            //yAxisID: `${type}-y${idx.toString()}`,
+            stack: `${type}-x0${isStacked ? "-stacked" : "y" + idx.toString()}`,
+            type: `${type}`,
+            backgroundColor: '',
+            borderColor: '',
+            yPos: idx,
+            datalabels: {
+              anchor: "end",
+            },
+          };
+        });
+      }
+    },
+
+    getPointDetails: (xPos: number, yPos: number): PointVal[] => {
+      console.log(type);
+      if (!_.isEmpty(legend)) {
+        return [
+          {
+            columnId: legend.id,
+            value: getDataForColumn(legend, dataArr)[yPos],
```

Comment:
why is this not xPos?

Reply:
xPos here can be anywhere around xAxis Points and it will more in case of slicing with color. yPos will help identifying which legend attribute we have clicked

Reply:
with xPos we can't identify the legend data here